test/e2e/upgrade_test.go

```diff
@@ -105,6 +111,94 @@ func testUpgradeVersion(t *testing.T) {
+			changeCommand := fmt.Sprintf("sc.exe config hybrid-overlay-node binPath=\\\"%s\\\"", newPath)
+			out, err = tc.runPowerShellSSHJob("hybrid-overlay-change", changeCommand, addr)
+			require.NoError(t, err, "error changing hybrid-overlay command")
+
+			require.NoError(t, tc.triggerWICDReconciliation(&node), "error triggering reconciliation")
+
+			// Wait until hybrid-overlay-node is returned to correct config
+			err = wait.Poll(retry.Interval, retry.Timeout, func() (bool, error) {
+				out, err = tc.runPowerShellSSHJob("hybrid-overlay-query2", queryCommand, addr)
+				if err != nil {
+					return false, nil
+				}
+				currentPath := finalLine(out)
+				if currentPath != originalPath {
+					log.Printf("waiting for hybrid-overlay service config to be reconciled, current value: %s",
+						currentPath)
+					return false, nil
+				}
+				return true, nil
+			})
+			require.NoError(t, err)
+		})
+	}
+}
+
+// TODO: Remove this function when https://issues.redhat.com/browse/WINC-736 is complete
+// triggerWICDReconciliation kicks off WICD instance reconciliation by changing the desiredVersionAnnotation to an
+// incorrect version, and then back to the correct version.
+func (tc *testContext) triggerWICDReconciliation(node *v1.Node) error {
+	patchData := fmt.Sprintf(`{"metadata":{"annotations":{"%s":"%s"}}}`, nodeconfig.DesiredVersionAnnotation,
+		"test")
+	_, err := tc.client.K8s.CoreV1().Nodes().Patch(context.TODO(), node.Name, types.MergePatchType,
+		[]byte(patchData), metav1.PatchOptions{})
+	if err != nil {
+		return errors.Wrap(err, "error patching node annotation")
+	}
+	patchData = fmt.Sprintf(`{"metadata":{"annotations":{"%s":"%s"}}}`, nodeconfig.DesiredVersionAnnotation,
+		node.Annotations[nodeconfig.DesiredVersionAnnotation])
+	_, err = tc.client.K8s.CoreV1().Nodes().Patch(context.TODO(), node.Name, types.MergePatchType,
+		[]byte(patchData), metav1.PatchOptions{})
+	if err != nil {
+		return errors.Wrap(err, "error patching node annotation back to expected")
+	}
+	return nil
+}
+
+// changeHybridOverlayCommandVerbosity will change the loglevel argument in the hybrid-overlay-node command to a
+// different level. It takes the full command path of the hybrid-overlay-node service, and returns the command with
+// altered arguments.
+func changeHybridOverlayCommandVerbosity(in string) (string, error) {
+	loglevelSplit := strings.Split(in, "loglevel ")
+	if len(loglevelSplit) != 2 {
+		return "", fmt.Errorf("exactly one instance of 'loglevel' must be present, found %d", len(loglevelSplit))
+	}
+	if len(loglevelSplit[1]) == 0 {
+		return "", fmt.Errorf("loglevel value not found")
+	}
+	originalLogLevel, err := strconv.Atoi(loglevelSplit[1][0:1])
+	if err != nil {
+		return "", err
+	}
+	var newLogLevel int
+	if originalLogLevel == 0 {
+		newLogLevel = originalLogLevel + 1
+	} else {
+		newLogLevel = originalLogLevel - 1
+	}
+	return fmt.Sprintf("%s%d%s", loglevelSplit[0], newLogLevel, loglevelSplit[1][1:]), nil
```

Comment:
Looks like the logic depends on `loglevel` always being the last arg and a single digit. Consider using a regex to simplify all this. Example: https://go.dev/play/p/_6LYdxD2-en